src/support/generation.c

```diff
@@ -156,25 +156,30 @@ __wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
                 __wt_sleep(0, 10);
 
             /*
-             * If we wait for more than a minute, log the event. In DIAGNOSTIC mode, abort if we
-             * ever wait more than 3 minutes, that's forever.
+             * If we wait for more than a minute, log the event. In diagnostic mode, abort if we
+             * ever wait more than the configured timeout.
              */
             if (minutes == 0) {
                 minutes = 1;
                 __wt_epoch(session, &start);
             } else {
                 __wt_epoch(session, &stop);
                 time_diff_ms = WT_TIMEDIFF_MS(stop, start);
-#define WT_GEN_DRAIN_TIMEOUT_MIN 4
+
                 if (time_diff_ms > minutes * WT_MINUTE * WT_THOUSAND) {
                     __wt_verbose_notice(session, WT_VERB_GENERATION,
                       "%s generation drain waited %" PRIu64 " minutes", __gen_name(which), minutes);
                     ++minutes;
-                    WT_ASSERT(session, minutes < WT_GEN_DRAIN_TIMEOUT_MIN);
                 }
-                /* Enable extra logs 20ms before timing out. */
-                else if (!verbose_timeout_flags &&
-                  time_diff_ms > (WT_GEN_DRAIN_TIMEOUT_MIN * WT_MINUTE * WT_THOUSAND - 20)) {
+
+                /* If no timeout is configured, there is nothing else to do. */
+                if (conn->gen_drain_timeout_ms == 0)
```

Comment:
is this check necessary?

Reply:
I thought it would help readability. We only want to do specific actions (see lines below) when a timeout has been configured (i.e greater than 0).